Content.Server/GameObjects/EntitySystems/StationEvents/RadiationPulseSystem.cs

```diff
@@ -45,10 +23,12 @@ public override void Update(float frameTime)
                 foreach (var entity in EntityManager.GetEntitiesInRange(ent.Transform.Coordinates, comp.Range, true))
                 {
                     if (entity.Deleted) continue;
-
-                    foreach (var radiation in entity.GetAllComponents<IRadiationAct>())
+                    if (comp.InRangeUnOccluded(entity, range: comp.Range))
                     {
-                        radiation.RadiationAct(frameTime, comp);
+                        foreach (var radiation in entity.GetAllComponents<IRadiationAct>())
+                        {
+                            radiation.RadiationAct(frameTime, comp);
+                        }
```

Comment:
I ended up profiling this as it looked hella expensive and well...

![image](https://user-images.githubusercontent.com/31366439/104072638-d7279b80-525f-11eb-8f7d-c111fb787111.png)

Raycasts ain't cheap. What you should do probably is get all of the IRadiationAct components up front and check if any exist to skip the raycast. Also maybe just do the raycast every half-second or something rather than every tick.